services/src/test/java/org/exoplatform/onlyoffice/OnlyofficeEditorServiceTest.java

```diff
@@ -55,6 +55,58 @@ public void beforeClass() {
     this.sessionProviderService = getContainer().getComponentInstanceOfType(SessionProviderService.class);
   }
 
+  /**
+   * Test add file preferences
+   */
+  @Test
+  public void testAddFilePreferences() throws Exception {
+    // Given
+    Node node = createDocument("Test Document.docx", "nt:file", "testContent", true);
+    String[] initialMixinNodeTypes = ((NodeImpl) node).getMixinTypeNames();
+
+    // When
+    editorService.addFilePreferences(node, "john", "path");
+
+    // Then
+    assertFalse(ArrayUtils.contains(initialMixinNodeTypes, "eoo:onlyofficeFile"));
+
+    String[] newMixinNodeTypes = ((NodeImpl) node).getMixinTypeNames();
+    assertTrue(ArrayUtils.contains(newMixinNodeTypes, "eoo:onlyofficeFile"));
```

Comment:
We should also assert the content of the node : check child existence "eoo:preferences", its children naed after the user id, and its path property.